feat: Implemented create and delete postings for team matching api

vvyn opened the pull request:
Added
- create Posting api function
- delete Posting api function

vvyn committed:
feat(postings): Create and delete postings

vvyn committed:
feat(postings): Miminal changes to create and delete posting functions

vvyn committed:
Merge branch 'develop' into feat/create-and-delete-posting-api

DedsecKnight commented on pages/api/posting/delete.tsx:
```diff
@@ -0,0 +1,61 @@
+  postingId: string;
+}
+
+async function deletePosting(req: NextApiRequest, res: NextApiResponse) {
```
I think we'll need to have a way to make sure that only post author can delete posting.

vvyn replied:
I added the changes, but wanted to let you know in case there are problems down the line with eslint, my branches are a bit messed up so I copied and pasted my files and committed them here through the github website instead of commiting and pushing from vs code.

vvyn committed:
Update create.tsx

vvyn committed:
feat(postings): Add verification of author for posting deletion

vvyn committed:
fix(matching): Rename posting directory and associated files

Rename posting directory to matching and create and delete files to
createPosting and deletePosting

vvyn committed:
Merge branch 'develop' into feat/create-and-delete-posting-api
